### PR TITLE
yarn2nix: allow running scripts

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
@thatsmydoing

File: pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix [hunk 1]
```diff
@@ -141,7 +141,7 @@ in rec {
 
         ${workspaceDependencyLinks}
 
-        yarn install ${lib.escapeShellArgs (defaultYarnFlags ++ yarnFlags)}
+        yarn install ${lib.escapeShellArgs (defaultYarnFlags ++ lib.optional "--ignore-scripts" ++ yarnFlags)}
```
Comment (thatsmydoing) [inline pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:144]: This looks wrong, should it be `defaultYarnFlags ++ (lib.optionals ignoreScripts ["--ignore-scripts"]) ++ yarnFlags`?

File: pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix [hunk 2]
```diff
@@ -413,7 +413,7 @@ in rec {
     # we import package.json from the unfiltered source
     packageJSON = ./package.json;
 
-    yarnFlags = defaultYarnFlags ++ ["--production=true"];
+    yarnFlags = defaultYarnFlags ++ [ "--ignore-scripts" "--production=true" ];
```
Comment (thatsmydoing) [inline pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:416]: This shouldn't be necessary since you default `ignoreScripts` to true.

**Review by thatsmydoing — COMMENTED** [review]
I feel like this is just addressing the symptom rather than the actual problem. I think there should be a way to override the flags as a whole rather than special casing a particular flag.

I think it's probably acceptable. I can imagine some situations in which we'd want to disable `--ignore-engines` too. In fact, I think having it on by default is wrong and users should opt in to ignoring engine version mismatches or they should instead override the nodejs version used by yarn2nix.

File: pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:144]: ``lib.optionals ignoreScripts ["ignore-scripts"]`` and lib.optional ignoreScripts "--ignore-scripts"`` is the same thing and you can chain lib.optional without ``()``.

File: pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:416]: This is a different function which does not take any arguments.

**Comment by SuperSandro2000** [comment]
> I feel like this is just addressing the symptom rather than the actual problem. I think there should be a way to override the flags as a whole rather than special casing a particular flag.

Overwriting defaultYarnFlags as whole is bad because you must always supply ``--offline`` or it won't work. 

> In fact, I think having it on by default is wrong and users should opt in to ignoring engine version mismatches or they should instead override the nodejs version used by yarn2nix.

This just ends up in micromanaging a setting which has almost no impact on our packaging. Developers like to pin this or set to overly strict version ranges with no reason even if older/newer nodejs version work totally fine. Then we must change the nodejs version and have even less cache hits because all modules are duplicated across versions.

File: pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:144]: Oh, you're right. Ommiting () only works when assigning to a var, not as function argument.

Comment (SuperSandro2000) [reply pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix]: I forgot the first argument 🤦🏼

File: pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix (hunk 2 above)
Comment (thatsmydoing) [reply pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix:416]: Isn't it the same? This function is `mkYarnPackage` which forwards `yarnFlags` to `mkYarnModules`?

**Pull request merged**